Source/WebCore/platform/glib/SystemSettingsGLib.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2024 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "SystemSettingsGLib.h"
+
+#include "FontRenderOptions.h"
+#include "RenderTheme.h"
+
+namespace WebCore {
+
+SystemSettingsGLib& SystemSettingsGLib::singleton()
+{
+    static NeverDestroyed<SystemSettingsGLib> settings;
+    return settings;
+}
+
+SystemSettingsGLib::SystemSettingsGLib()
+{
+}
+
+void SystemSettingsGLib::applySettings(SettingsStateGLib&& state)
+{
+    bool shouldUpdateStyle = false;
+    if (m_settings.themeName != state.themeName) {
```

Comment:
The state members are optional, so if the member is nulltopt it means the value hasn't changed. So, if we have a theme name already and it hasn't changed, this condition will be true, but we don't want to change the setting at all. I don't think we need to compare the values, if it has changed, the value should always be different.